midend/lib/Conversion/ConvOptimization/ConvOptimize.cpp

```diff
@@ -63,88 +69,198 @@ class ConvOptimizePattern : public ConversionPattern {
+                                                              loc, vecTy, cf0);
+                                                  elseBuilder.create<
+                                                      affine::AffineYieldOp>(
+                                                      loc, emptyVec);
+
+                                                  iList.push_back(i);
+                                                  fList.push_back(
+                                                      protectedF->getOpResult(
+                                                          0));
+                                                }
+                                              }
+                                              Value lastResult =
+                                                  builder
+                                                      .create<memref::LoadOp>(
+                                                          loc, buffer, c0);
+                                              for (int i = 0; i < kernelM;
+                                                   ++i) {
+                                                for (int j = 0; j < kernelN;
+                                                     ++j) {
+                                                  lastResult = builder.create<
+                                                      vector::FMAOp>(
+                                                      loc, vecTy,
+                                                      iList[i * kernelN + j],
+                                                      fList[i * kernelN + j],
+                                                      lastResult);
+                                                }
+                                              }
+
+                                              builder.create<memref::StoreOp>(
+                                                  loc, lastResult, buffer, c0);
+                                            });
+                                      });
+                                });
+
+                            Value reduceVec =
+                                builder.create<memref::LoadOp>(loc, buffer, c0);
+                            Value reducedRes =
+                                builder.create<vector::ReductionOp>(
+                                    loc, vector::CombiningKind::ADD, reduceVec);
+                            Value bias = builder.create<memref::LoadOp>(
+                                loc, output, ValueRange{ivA, ivB, ivC, ivD});
+                            Value addRes = builder.create<arith::AddFOp>(
+                                loc, bias, reducedRes);
+                            builder.create<memref::StoreOp>(
+                                loc, addRes, output,
+                                ValueRange{ivA, ivB, ivC, ivD});
+                          });
+                    });
               });
```

Comment:
Is the purpose of modifying this code for beautifying/formatting? Would the original code look better?